### DISCUSSION TITLE
How to validate Gzip files

### DISCUSSION THREAD

**Discussion by gopimanikandan** [icsharpcode/SharpZipLib#521] [Q&A]
There is no option to validate the Gzip file is corrupted or not. ZipFile has the option to TestArchive but Gzip couldn't.
Please let me know if there is any option to validate the Gzip file before extract it.

**Comment by piksel** [comment]
The difference between the Zip format and the GZip format is that Zip is not just compression, it also has a "file system" of sorts. That is what `TestArchive` checks when you run it. GZip is just a compressed stream of bytes, so there is really nothing to check except if it decompresses correctly.
If you want to do a pre-check for the contents you could just continuously `Read()` from it and check for any exceptions, that way you can verify that the GZip stream is OK without committing anything to disk/memory.
